lib/Data/Chronicle/Reader.pm

```diff
@@ -205,6 +205,35 @@ sub get_for_period {
     return \@result;
 }
 
+=head3 C<< my $data = get_history("category1", "name1", 1) >>
+
+Query Pg archive for the data under "category1::name1" at the provided number of revisions in the past.
+
+=cut
+
+sub get_history {
+    my $self     = shift;
+    my $category = shift;
+    my $name     = shift;
+    my $rev      = shift;
+
+    die "Revision must be >= 0" if $rev < 0;
```

Comment:
I don't know if this can happen in this point, but If `$rev = undef` this check will pass and break the result.